index.js

```diff
@@ -1,9 +1,32 @@
 'use strict';
+const URL = require('url').Url;
 const urlRegex = require('url-regex');
 const normalizeUrl = require('normalize-url');
+const unescape = require('unescape');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	opts = opts || {};
+	const ret = new Set();
+
+	function _extractQueryParams(url) {
+		const qs = new URL().parse(url, true).query;
+		for (const key in qs) {
```

Comment:
Use a `for-of` loop. Then you don't have to use `Object.prototype.hasOwnProperty.call(qs, key)`.

Reply:
the returned object is an object not iterable therefor not applicable here

Reply:
```js
for (const key of Object.keys(qs)) {
```

Reply:
ok will do

Reply:
done